src/common/src/com/microsoftopentechnologies/tooling/msservices/helpers/azure/rest/AzureCertificateHelper.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright 2014 Microsoft Open Technologies Inc.
+ * <p/>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.microsoftopentechnologies.tooling.msservices.helpers.azure.rest;
+
+import com.microsoftopentechnologies.tooling.msservices.helpers.NotNull;
+import com.microsoftopentechnologies.tooling.msservices.helpers.Nullable;
+import com.microsoftopentechnologies.tooling.msservices.helpers.azure.AzureCmdException;
+import com.microsoftopentechnologies.tooling.msservices.helpers.azure.rest.RestServiceManager.ContentType;
+import com.microsoftopentechnologies.tooling.msservices.helpers.azure.rest.RestServiceManager.HttpsURLConnectionProvider;
+
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLSocketFactory;
+
+public class AzureCertificateHelper {
+    @NotNull
+    public static String executeRequest(@NotNull String managementUrl,
+                                        @NotNull String path,
+                                        @NotNull ContentType contentType,
+                                        @NotNull String method,
+                                        @Nullable String postData,
+                                        @NotNull SSLSocketFactory sslSocketFactory,
+                                        @NotNull RestServiceManager manager)
+            throws AzureCmdException {
+        HttpsURLConnectionProvider sslConnectionProvider = getHttpsURLConnectionProvider(sslSocketFactory, manager);
+
+        return manager.executeRequest(managementUrl, path, contentType, method, postData, sslConnectionProvider);
```

Comment:
Actually, never mind. I see now that this abstraction exists so that implementations of `RestServiceManager` can acquire a `HttpsURLConnection` object without needing to know whether they are dealing with an SSL client certificate or an Azure AD token.